### PR TITLE
Fix browser omnibar Option-Backspace URL deletion

### PR DESCRIPTION
**Pull request by lawrencecchen**
## Summary

- Intercepts plain Option-Backspace in the browser omnibar before AppKit treats a full hostname as one word.
- Deletes back to URL-aware boundaries such as `.`, `/`, `?`, `#`, `&`, `=`, `@`, `:`, and whitespace.
- Adds regression coverage for `news.ycombinator.com` becoming `news.ycombinator.`.

## Testing

- PASS: `./scripts/lint-pbxproj-test-wiring.sh`
- PASS: `./scripts/reload.sh --tag urlwd`
- BLOCKED: AWS focused `xcodebuild ... -only-testing:cmuxTests/OmnibarStateMachineTests/testOptionBackspaceDeletesOnlyFinalURLHostLabel test`
  - Red attempt on `c76a54cd5` did not reach the test because the AWS builder ran out of disk while compiling the existing Rust FFI build step.
  - Green attempt on `41193e1ff` compiled but the first test run failed before execution because `testmanagerd` was unavailable in the non-GUI launch context; the GUI-context retry then hit AWS disk exhaustion while resolving Sentry artifacts.

## Dogfood

- Tagged build: `urlwd`
- Expected behavior: focus the cmux browser omnibar with `news.ycombinator.com`, press Option-Backspace once, and the field should read `news.ycombinator.` instead of becoming empty.

---
<a href="https://app.blacksmith.sh/manaflow-ai/codesmith/cmux/pr/4651"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1782118840&installation_id=133855650&pr_number=4651&repository=manaflow-ai%2Fcmux&return_to=https%3A%2F%2Fgithub.com%2Fmanaflow-ai%2Fcmux%2Fpull%2F4651&signature=64379441ca0e57fde93b24bad462314b4a86de7c408ba26779fc613801f187d3"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>@codesmith</code> with what you need. Autofix is disabled.</sup>

---

> [!NOTE]
> **Medium Risk**
> Changes key handling and text mutation for the browser omnibar’s field editor, which can impact deletion semantics, selection, and undo behavior. Adds URL-aware parsing logic with several edge cases, increasing risk of regressions in text editing interactions.
> 
> **Overview**
> Fixes browser omnibar word-deletion so **Option-Backspace / `deleteWordBackward` become URL-aware** for single-token URLs, deleting back to boundaries like `.`, `/`, `?`, `#`, `&`, `=`, `@`, and `:` instead of AppKit treating the whole hostname as one word.
> 
> Adds `omnibarTextAfterDeletingURLWordBackward` (returns updated text + deleted/selection ranges) and wires it into both inline-completion deletion and plain editing by intercepting Option-Backspace and programmatically applying the deletion while preserving selection and undo.
> 
> Expands tests to cover URL host-label deletion, mid-string deletion without double separators, consuming separator runs (e.g. `https://`), command-based deletion, and fallthrough to native behavior for non-URL text.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit a9476f314a14d0922fd713b34692156c34fa8c66. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

---
## Summary by cubic
Fixes the browser omnibar so Option-Backspace and `deleteWordBackward` delete one URL segment using URL-aware boundaries in both plain text and inline-completion states. Consumes separator runs (e.g., `://`), avoids double separators, preserves undo/selection, and falls back to native deletion for non-URL text.

- **Bug Fixes**
  - Applies URL-aware deletion when inline completion is active or not; stops at `.`, `/`, `?`, `#`, `&`, `=`, `@`, `:`, or whitespace (caret only).
  - Routes key events and `deleteWordBackward` to the URL-aware path; consumes adjacent separator runs and removes lone delimiters; preserves caret/selection and undo; adds tests for inline completion, mid-string, after separators, scheme-run, command routing, and non-URL fallback.

<sup>Written for commit 55e4389ca8b662e1cb77897a9caa32667e18b176. Summary will update on new commits. <a href="https://cubic.dev/pr/manaflow-ai/cmux/pull/4651?utm_source=github">Review in cubic</a></sup>

## Summary by CodeRabbit

* **New Features**
  * Omnibar: Option+Backspace and word-delete now remove URL segments cleanly (host labels and trailing separators), preserve caret/selection and undo history, and clear inline suggestions when deleting URL parts before typed text.

* **Tests**
  * Added unit tests covering Option+Backspace and command-based word deletion, selection/undo correctness, and URL-separator edge cases (plain and inline omnibar modes).

[Review Change Stack](https://app.coderabbit.ai/change-stack/manaflow-ai/cmux/pull/4651?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: Sources/Panels/BrowserPanelView.swift [hunk 1]
```diff
@@ -3141,6 +3141,53 @@ func omnibarPrefixAfterDeletingTrailingWord(from text: String) -> String {
...
+
+    var deletionStart = caret
+    while deletionStart > 0,
+          !omnibarURLWordBoundaryCharacter(in: nsText, at: deletionStart - 1) {
+        deletionStart -= 1
+    }
+
+    guard deletionStart < caret else { return nil }
+    let deletionRange = NSRange(location: deletionStart, length: caret - deletionStart)
+    let updated = nsText.replacingCharacters(in: deletionRange, with: "")
```
[the remark this answers is not in the document]
Comment (lawrencecchen) [reply Sources/Panels/BrowserPanelView.swift:3211]: Fixed in commits f59e91bcb and 3f80697e6. Added the mid-string regression test and consume one adjacent URL separator so news.ycombinator.com no longer becomes news..com.

*— Claude Code*

File: Sources/Panels/BrowserPanelView.swift [hunk 2]
```diff
@@ -4284,6 +4339,28 @@ struct OmnibarTextFieldRepresentable: NSViewRepresentable {
...
+                selectedRange: editor.selectedRange()
+            ) else {
+                return false
+            }
+
+            isProgrammaticMutation = true
+            editor.string = deletion.text
+            parentField?.stringValue = deletion.text
+            editor.setSelectedRange(deletion.selectedRange)
+            isProgrammaticMutation = false
```
[the remark this answers is not in the document]
Comment (lawrencecchen) [reply Sources/Panels/BrowserPanelView.swift]: Fixed in c66dc245d. The URL-aware edit now goes through shouldChangeText/didChangeText with the text storage replacement, and the test harness asserts the undo manager can restore the deleted URL label.

*— Claude Code*

File: Sources/Panels/BrowserPanelView.swift [hunk 3]
```diff
@@ -3141,6 +3141,53 @@ func omnibarPrefixAfterDeletingTrailingWord(from text: String) -> String {
...
+          selectedRange.length == 0,
+          selectedRange.location <= textLength,
+          selectedRange.location > 0 else {
+        return nil
+    }
+
+    let caret = selectedRange.location
+    guard !omnibarURLWordBoundaryCharacter(in: nsText, at: caret - 1) else {
+        return nil
+    }
```
[the remark this answers is not in the document]
Comment (lawrencecchen) [reply Sources/Panels/BrowserPanelView.swift:3180]: Fixed in 104eae841 and 7d9321253. Added coverage for deleteWordBackward: and repeated deletion after a URL separator, then routed the field-editor command path through the same URL-aware helper.

*— Claude Code*
